tests/coverage/drop_trait.cov-map

```diff
@@ -7,15 +7,13 @@ Number of file 0 mappings: 1
 - Code(Counter(0)) at (prev + 9, 5) to (start + 2, 6)
 
 Function name: drop_trait::main
-Raw bytes (26): 0x[01, 01, 01, 05, 00, 04, 01, 0e, 01, 05, 0c, 05, 06, 09, 01, 16, 00, 02, 06, 04, 0b, 03, 05, 01, 00, 02]
+Raw bytes (24): 0x[01, 01, 00, 04, 01, 0e, 01, 05, 0c, 05, 06, 09, 01, 16, 00, 02, 06, 04, 0b, 01, 05, 01, 00, 02]
 Number of files: 1
 - file 0 => global file 1
-Number of expressions: 1
-- expression 0 operands: lhs = Counter(1), rhs = Zero
+Number of expressions: 0
 Number of file 0 mappings: 4
 - Code(Counter(0)) at (prev + 14, 1) to (start + 5, 12)
 - Code(Counter(1)) at (prev + 6, 9) to (start + 1, 22)
 - Code(Zero) at (prev + 2, 6) to (start + 4, 11)
-- Code(Expression(0, Add)) at (prev + 5, 1) to (start + 0, 2)
-    = (c1 + Zero)
```

Comment:
Any explanation of why this (essentially `Counter(1)`) is turned into `Counter(0)` instead?
This same pattern turns up in a bunch of files.

Reply:
In general an expression can be lowered to zero due to some constants. For example, mir of this test function is 
```
    coverage ExpressionId(0) => Expression { lhs: Counter(0), op: Subtract, rhs: Counter(1) };
    coverage ExpressionId(1) => Expression { lhs: Counter(1), op: Add, rhs: Expression(0) };

    bb0: {
        Coverage::CounterIncrement(0);
        // ... 
        _3 = const true;
        switchInt(move _3) -> [0: bb4, otherwise: bb1];
    }

    bb1: {
        Coverage::CounterIncrement(1);
        // ... 
    }

    bb4: {
        Coverage::ExpressionUsed(0);
        // ... 
    }
```
Since `_3` is always true, we have `Counter 1 == Counter 0` and `Expression 0` can be lowered to `Zero`.
Then `Expression 1 := Counter 1 + Expression 0`, and we can see `(c1 + Zero)` here in previous.

This patch simplifies expressions before any expression is lowered to `Zero`. Therefore we get `Expression 1:= Counter 1 + Expression 0 = Counter 1 + (Counter 0 - Counter 1) = Counter 0` first. Hence later it is not related to `c1` or `Zero`.

Reply:
Ah, I see. So in other words:

- Without expression simplification, this would have been `c1 + (c0 - c1)`.
  - Then the block containing `(c0 - c1)` is removed by MIR opts, so coverage codegen replaces it with `Zero`, and we get `c1 + Zero`.
- With simplification, we just immediately get `c1`, since `c0` cancels itself out.